docs/multitenancy/proposal.md

```diff
@@ -0,0 +1,71 @@
+
+Each deployment of the operator will run in a separate namespace (rather than the current default `azureoperator-system`) and read its configuration from a secret named `azureoperatorsettings` in that namespace.
+If the operator is configured to use managed identity (`AZURE_USE_MI=1`), then the `AzureIdentity` and `AzureIdentityBinding` will be found in that namespace as well.
+Each instance of the operator will use the service account defined in its namespace.
+(This is all the same as the current behaviour of the operator.)
+
+A new configuration attribute `AZURE_TARGET_NAMESPACES` will control which namespaces that operator will monitor.
+This should be a comma-separated list of names, which could include the namespace containing the operator deployment (but doesn't have to).
+Having no namespaces in the list or omitting the setting from the configuration will result in the operator watching all namespaces for ASO resources as it does now.
+
+// TODO: How to ensure that each namespace is managed by at most one operator? Having different operators (with potentially different credentials) fighting over a single resource would be messy. Similarly having both a global ASO and per-namespace instances would cause problems.
+
+### CRDs and webhooks
+
+`CustomResourceDefinitions`, `ValidatingWebhookConfigurations` and `MutatingWebhookConfigurations` are cluster-wide so they can't be managed on a per-operator basis.
+In a multitenant configuration installing and managing the CRDs and webhooks will be a separate step from deploying operators (and will require cluster admin privileges).
+
+The conversion webhooks in the CRDs and the validating and mutating webhook configurations that are defined for different ASO resources will all point to the `azureoperator-webhook-service` service deployed in the `azureoperator-system` namespace (by default).
+The service will be backed by a deployment of the normal ASO operator running in a restricted webhook-only mode - it won't be watching any namespaces for Azure resources and won't have access to any Azure credentials.
+This mode will be triggered by the configuration setting `AZURE_WEBHOOK_ONLY=1`.
+With this set no other configuration values will be required, and no resource watches will be started - the operator will just respond to webhook requests from the API server.
+
+### RBAC
+
+At the moment installing a global instance of ASO creates a `ClusterRole` defining all of the permissions the operator requires on the different Azure resources (`azureoperator-manager-role`), and then a `ClusterRoleBinding` granting those permissions for resources in all namespaces (`azureoperator-manager-rolebinding`).
+A multitenant deployment won't have the global `ClusterRoleBinding`; instead each managed namespace N will need to contain a `RoleBinding` connecting the global `ClusterRole` with the service account of the operator that will manage N's resources.
```

Comment:
We'll have to talk about how to make management of this easy... or maybe just "easier" since it's inherently somewhat complicated

Reply:
Definitely - I'm vaguely picturing a tool to deploy it that would take the list of namespaces as input somehow and create the rolebindings at the same time as creating the deployment.